pkg/kademlia/dialer.go

```diff
@@ -109,6 +109,23 @@ func (dialer *Dialer) FetchPeerIdentity(ctx context.Context, target pb.Node) (pI
 	return identity.PeerIdentityFromPeer(p)
 }
 
+// FetchInfo connects to a node address and returns its node info.
+func (dialer *Dialer) FetchInfo(ctx context.Context, target pb.Node) (*pb.InfoResponse, error) {
+	if !dialer.limit.Lock() {
+		return nil, context.Canceled
+	}
+	defer dialer.limit.Unlock()
+
+	conn, err := dialer.dial(ctx, target)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := conn.client.RequestInfo(ctx, &pb.InfoRequest{})
+
+	return resp, errs.Combine(err, conn.disconnect())
```

Comment:
Since we established connection we can still return the node id. So the method would be:

```
func (dialer *Dialer) FetchInfo(ctx context.Context, target pb.Node) (storj.NodeID, *pb.InfoResponse, error)

    p  := &peer.Peer{}
    pCall := grpc.Peer(p)
    resp, err = conn.client.RequestInfo(ctx, &pb.InfoRequest{}, pCall)
    id ... := identity.PeerIdentityFromPeer(p)
   return id, resp, ...
}
```

Reply:
OK. This sounds good. I will bring the Node ID back to the response.

Reply:
Done.